### PR TITLE
[ws-daemon] log when ready probe fails

### PR DESCRIPTION
**Pull request by sagor999**
## Description

Make sure that we log when ready probe fails. Currently it is never reported anywhere.

## Related Issue(s)

Will help debug https://github.com/gitpod-io/gitpod/issues/10081

## Release Notes

```release-note
[ws-daemon] add log entry when ready probe fails
```

### CODE REVIEW

**Comment by roboquat** [comment]
@sagor999: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/ws-daemon/pkg/daemon/daemon.go [hunk 1]
```diff
@@ -202,16 +203,22 @@ func (d *Daemon) Stop() error {
 func (d *Daemon) ReadinessProbe() func() error {
 	return func() error {
 		if d.hosts != nil && !d.hosts.DidUpdate() {
-			return fmt.Errorf("host controller not ready yet")
+			err := fmt.Errorf("host controller not ready yet")
+			log.Errorf("ready probe failure: %v", err)
```
Comment (csweichel) [inline components/ws-daemon/pkg/daemon/daemon.go:207]:
```suggestion
			log.WithError(err).Errorf("readiness probe failure")
```

File: components/ws-daemon/pkg/daemon/daemon.go [hunk 2]
```diff
@@ -202,16 +203,22 @@ func (d *Daemon) Stop() error {
...
+			return err
 		}
 
 		isContainerdReady, err := d.dispatch.Runtime.IsContainerdReady(context.Background())
 		if err != nil {
-			return fmt.Errorf("containerd error: %v", err)
+			err := fmt.Errorf("containerd error: %v", err)
+			log.Errorf("ready probe failure: %v", err)
```
Comment (csweichel) [inline components/ws-daemon/pkg/daemon/daemon.go:214]:
```suggestion
			log.WithError(err).Errorf("readiness probe failure: containerd error")
```

File: components/ws-daemon/pkg/daemon/daemon.go [hunk 3]
```diff
@@ -202,16 +203,22 @@ func (d *Daemon) Stop() error {
...
+			return err
 		}
 
 		if !isContainerdReady {
-			return fmt.Errorf("containerd is not ready")
+			err := fmt.Errorf("containerd is not ready")
+			log.Errorf("ready probe failure: %v", err)
```
Comment (csweichel) [inline components/ws-daemon/pkg/daemon/daemon.go:220]:
```suggestion
			log.WithError(err).Error("readiness probe failure")
```

**Comment by sagor999** [comment]
@csweichel PTAL :heart:

**Review by csweichel — APPROVED** [review]

**Pull request merged**